daemon/daemon.go

```diff
@@ -196,16 +196,15 @@ func (daemon *Daemon) RegistryHosts(host string) ([]docker.RegistryHost, error)
 	}
 	conf := daemon.registryService.ServiceConfig().IndexConfigs
 	for k, v := range conf {
-		c := resolverconfig.RegistryConfig{}
+		c := m[k]
 		if !v.Secure {
 			t := true
 			c.PlainHTTP = &t
 			c.Insecure = &t
```

Comment:
Perhaps we have checks elsewhere, but does this mean we allow `docker.io` to be used with "insecure"?

Reply:
It shouldn't because `docker.io` is always overriden with `Secure: true` entry: https://github.com/moby/moby/blob/aacd100a70ec3297e0480e9d1f2c8ba9d564bf82/registry/config.go#L236-L241

but... I think it _could_ be worked around if you explicitly use `registry-1.docker.io`? But it might get translated to `docker.io` before this is used, so perhaps it's not actually possible.